middle_end/flambda2/to_cmm/to_cmm_expr.ml

```diff
@@ -416,8 +474,10 @@ and let_expr0 env res let_expr (bound_pattern : Bound_pattern.t)
       let wrap, env, res =
         Env.flush_delayed_lets ~mode:Branching_point env res
       in
-      let body, res = expr env res body in
-      wrap (C.sequence update body), res)
+      let body, body_free_vars, res = expr env res body in
+      let free_vars = Backend_var.Set.union update.free_vars body_free_vars in
```

Comment:
The binding of `update` is just outside the diff so I can't put a comment on it, but I think it would be nice to explicitly add a record pattern there (it will help if we need to add more fields to the record).

Reply:
done